presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/AbstractRowEncoder.java

```diff
@@ -201,6 +214,21 @@ protected void appendSqlTimestampWithTimeZone(SqlTimestampWithTimeZone value)
         throw new UnsupportedOperationException(format("Unsupported type '%s' for column '%s'", value.getClass().getName(), columnHandles.get(currentColumnIndex).getName()));
     }
 
+    protected void appendArray(List<Object> value)
+    {
+        throw new UnsupportedOperationException(format("Unsupported type '%s' for column '%s'", value.getClass().getName(), columnHandles.get(currentColumnIndex).getName()));
+    }
+
+    protected void appendMap(Map<Object, Object> value)
+    {
+        throw new UnsupportedOperationException(format("Unsupported type '%s' for column '%s'", value.getClass().getName(), columnHandles.get(currentColumnIndex).getName()));
+    }
+
+    protected void appendRow(List<Object> value)
+    {
+        throw new UnsupportedOperationException(format("Unsupported type '%s' for column '%s'", value.getClass().getName(), columnHandles.get(currentColumnIndex).getName()));
```

Comment:
you could extract a method in separate PR, as this would change the entire convention in this clas